# title_url 돌려 feature 뽑기

In [1]:
import requests
from bs4 import BeautifulSoup
import json
import pandas as pd

In [2]:
url_df = pd.read_csv('./resource/url_df.csv')
url_df

title_url    code
0                    spotlight-2015  mnp3sa
1      wiheomhan-somun-jjirasi-2013  mizwpr
2              before-midnight-2013  mkrc6e
3                        iyagi-2012  m3vcz7
4                transcendence-2014  mef2j2
5                 inganjungdog-2013  mkpnzb
6             the-hunger-games-2014  mfvxi2
7                   the-intern-2015  mlqosw
8               byuti-insaideu-2014  mndtm0
9             geomeun-sajedeul-2015  mtb5m7
10                 naebujadeul-2014  mtndts
11                 silent-hill-2006  mluxpt
12                 the-martian-2015  mc4mas
13     the-time-travelers-wife-2009  mwfm8t
14                       gundo-2012  mnerjj
15                    sin-city-2005  mo0weh
16           mudeomggaji-ganda-2013  muiysd
17                  big-hero-6-2014  mxy831
18                        lucy-2014  my1nf4
19             the-bucket-list-2007  mjoctb
20              guns-and-talks-2001  m4g420
21                  amsal-gaje-2015  mfbyqj
22                    beterang-2014  m0e0ng
23             the-maze-runner-2014  mitzzb
24                       d-war-2007  m0g5q8
25            geoncughaggaeron-2012  mi2x6x
26                 august-rush-2007  mfk8wd
27              bruce-almighty-2003  mp8xml
28                lust-caution-2007  mp0bz7
29                pulp-fiction-1994  ml4cgi
..                              ...     ...
514        gwanghae-wangi-doen-2012  motei7
515      7beonbangyi-seonmul-2012-1  mw4vzn
516                the-thieves-2012  m2vxj4
517     spirit-of-jeet-keun-do-2004  mv3b4e
518           71-into-the-fire-2010  mco8ts
519               spider-man-2-2004  mvqagi
520  qian-toqian-xun-falseshen-2001  mvzdp0
521       the-butterfly-effect-2004  mdoyxv
522             shutter-island-2010  m0gk77
523            the-truman-show-1998  mkmevg
524                    if-only-2004  mxmz5o
525        joint-security-area-2000  m35387
526       the-bourne-supremacy-2004  miogpj
527                   haunters-2010  mksqaw
528                beulraindeu-2011  ms0tq6
529         the-school-of-rock-2003  mywkgp
530         500-days-of-summer-2009  mrcy2s
531   the-shawshank-redemption-1994  m32kz6
532                love-letter-1995  mkecxr
533      beomjoewayi-jeonjaeng-2011  mh7hlp
534      the-devil-wears-prada-2006  mthg0d
535                       moss-2010  m2r7cf
536                 real-steel-2011  mash3e
537                       leon-1994  m4phlv
538         naega-salinbeomida-2012  mj6jfk
539                    malaton-2005  mwq1h6
540                the-classic-2003  mq867z
541                 iron-man-3-2013  mm8zqe
542            my-little-bride-2004  mm2fh0
543          all-about-my-wife-2012  mk035p

[544 rows x 2 columns]

In [4]:
for title in url_df['title_url']:
    print(title)

spotlight-2015
wiheomhan-somun-jjirasi-2013
before-midnight-2013
iyagi-2012
transcendence-2014
inganjungdog-2013
the-hunger-games-2014
the-intern-2015
byuti-insaideu-2014
geomeun-sajedeul-2015
naebujadeul-2014
silent-hill-2006
the-martian-2015
the-time-travelers-wife-2009
gundo-2012
sin-city-2005
mudeomggaji-ganda-2013
big-hero-6-2014
lucy-2014
the-bucket-list-2007
guns-and-talks-2001
amsal-gaje-2015
beterang-2014
the-maze-runner-2014
d-war-2007
geoncughaggaeron-2012
august-rush-2007
bruce-almighty-2003
lust-caution-2007
pulp-fiction-1994
sinsegye-2012
frozen-2013
kick-ass-2010
robot-2004
king-kong-2005
avatar-2009
the-notebook-2004
myeongryang-hoeoribada-2012
deadpool-2014
the-kirishima-thing-2012
before-sunrise-1995
miao-su-5sentime-toru-2007
gone-girl-2014
midnight-in-paris-2011
the-grand-budapest-hotel-2014
mad-max-fury-road-2013
all-you-need-is-kill-2014
jurassic-park-1993
x-men-days-of-future-2014
no-country-for-old-men-2007
the-secret-service-2015
interstellar-2012
whiplash-2014

In [87]:
url_df[400:]

title_url    code
400                   iussaram-2012  mz5nqi
401          x-men-first-class-2011  mjl2w4
402      miracle-on-1st-street-2007  mqkqgy
403             the-front-line-2011  m4awac
404                     mother-2009  mgnets
405            ossaghan-yeonae-2011  mnbgb2
406                    harmony-2009  m54gng
407                hello-ghost-2010  moliec
408       harry-potter-and-the-2007  m73ery
409       harry-potter-and-the-2001  myweqh
410               ssanghwajeom-2008  mlfk1z
411          the-divine-weapon-2008  m885we
412         forever-the-moment-2007  mmq4ll
413               daensingkwin-2012  ml1j18
414               spider-man-3-2007  mh8gxy
415            my-tutor-friend-2003  m8d550
416                 the-chaser-2008  miipzb
417   pirates-of-the-caribbean-2007  mcctgd
418  the-lord-of-the-rings-the-2002  m3qylw
419                    titanic-1997  mc57ne
420                 my-teacher-2005  m0il00
421     mission-impossible-iii-2006  mknkjl
422                 yihyeongje-2010  momlxz
423                      tower-2012  mbqdkn
424          the-war-of-flower-2006  mp1qyn
425             les-miserables-2012  mi7d0g
426                  inception-2010  mevucc
427     jeon-woochi-the-taoist-2009  m8hh0b
428          war-of-the-arrows-2011  mzxgsr
429               transformers-2007  m1um48
..                              ...     ...
514        gwanghae-wangi-doen-2012  motei7
515      7beonbangyi-seonmul-2012-1  mw4vzn
516                the-thieves-2012  m2vxj4
517     spirit-of-jeet-keun-do-2004  mv3b4e
518           71-into-the-fire-2010  mco8ts
519               spider-man-2-2004  mvqagi
520  qian-toqian-xun-falseshen-2001  mvzdp0
521       the-butterfly-effect-2004  mdoyxv
522             shutter-island-2010  m0gk77
523            the-truman-show-1998  mkmevg
524                    if-only-2004  mxmz5o
525        joint-security-area-2000  m35387
526       the-bourne-supremacy-2004  miogpj
527                   haunters-2010  mksqaw
528                beulraindeu-2011  ms0tq6
529         the-school-of-rock-2003  mywkgp
530         500-days-of-summer-2009  mrcy2s
531   the-shawshank-redemption-1994  m32kz6
532                love-letter-1995  mkecxr
533      beomjoewayi-jeonjaeng-2011  mh7hlp
534      the-devil-wears-prada-2006  mthg0d
535                       moss-2010  m2r7cf
536                 real-steel-2011  mash3e
537                       leon-1994  m4phlv
538         naega-salinbeomida-2012  mj6jfk
539                    malaton-2005  mwq1h6
540                the-classic-2003  mq867z
541                 iron-man-3-2013  mm8zqe
542            my-little-bride-2004  mm2fh0
543          all-about-my-wife-2012  mk035p

[144 rows x 2 columns]

---
### 추가 feature dataframe
* 작업시간 약 10분걸림
* 전체로하니 418개 나옴 (director부분에서 에러)
* 인덱스 419번째 영화 : 반지의제왕2 에서 문제발생 => 이것만 따로 떼어내서 합침
* 실수때문에 418번쨰 영화 : 타이타닉도 따로 뗴어내서 합침

** 1. df1 : 0~417번째까지**

In [42]:
df1 = pd.DataFrame(columns=['DESC', '감독', '배우', '평가자수', '보고싶어요수',
                           '코멘트수', '별점분포'])

for title in url_df['title_url'][:418]:
    response = requests.get("https://watcha.net/mv/{title_url}".format(title_url=title))
    dom = BeautifulSoup(response.content, 'html.parser')
    
    director = dom.select('li.movie-person.director a.kor-name')[1].text # 감독
    actor_list = dom.select('li.movie-person.actor .kor-name')
    actors = []
    for actor in actor_list[1:6:2]:
        actor = actor.text
        actors.append(actor) # 배우 3명
    rate_count = dom.select('div.cumulative-count .rate')[0].text # 평가자 수
    wish_count = dom.select('div.cumulative-count .wish')[0].text # 보고싶어요 수
    review_count = dom.select('div.cumulative-count .review')[0].text # 코메트 수
    desc = dom.select('div.desc')[0].text   # 장르가 포함된 DESC
    rating_dist = dom.select_one('div.rating-chart-wrapper').get('data-distribution') # 분포

    df1.loc[len(df1)] = [desc, director, actors, rate_count, wish_count, review_count, rating_dist]

In [43]:
df1

DESC         감독  \
0       Spotlight, 2015, 미국, 드라마, 스릴러, 15세 관람가, 2시간 8분    토마스 맥카시   
1                   2013, 한국, 드라마, 범죄, 15세 관람가, 2시간 1분        김광식   
2    Before Midnight, 2013, 미국, 로맨스/멜로, 청소년 관람불가, 1...  리처드 링클레이터   
3     Killer Toon, 2013, 한국, 공포, 스릴러, 15세 관람가, 1시간 44분        김용균   
4    Transcendence, 2014, 미국, 영국, SF, 스릴러, 12세 관람가,...     월리 피스터   
5    Obsessed, 2014, 한국, 드라마, 로맨스/멜로, 청소년 관람불가, 2시간...        김대우   
6    The Hunger Games: Mockingjay - Part 1, 2014, 미...   프란시스 로렌스   
7           The Intern, 2015, 미국, 코미디, 12세 관람가, 2시간 1분    낸시 마이어스   
8    The Beauty Inside, 2015, 한국, 로맨스/멜로, 12세 관람가, ...        백종열   
9    The Priests, 2015, 한국, 스릴러, 드라마, 15세 관람가, 1시간 48분        장재현   
10    Inside Men, 2015, 한국, 범죄, 드라마, 청소년 관람불가, 2시간 10분        우민호   
11   Silent Hill, 2006, 캐나다, 일본, 미국, 프랑스, 공포, 스릴러, ...   크리스토프 갱스   
12     The Martian, 2015, 미국, SF, 액션, 12세 관람가, 2시간 22분     리들리 스콧   
13   The Time Traveler's Wife, 2009, 미국, 로맨스/멜로, 판타...  로베르트 슈벤트케   
14   KUNDO : Age of the Rampant, 2014, 한국, 액션, 15세 ...        윤종빈   
15        Sin City, 2005, 미국, 범죄, 액션, 청소년 관람불가, 2시간 3분     프랭크 밀러   
16      A Hard Day, 2013, 한국, 범죄, 액션, 15세 관람가, 1시간 51분        김성훈   
17   Big Hero 6, 2014, 미국, 애니메이션, 액션, 코미디, 가족, SF, ...   크리스 윌리엄스   
18          LUCY, 2014, 미국, 프랑스, 액션, 청소년 관람불가, 1시간 30분       뤽 베송   
19   The Bucket List, 2007, 미국, 드라마, 코미디, 12세 관람가, ...      롭 라이너   
20     Guns & Talks, 2001, 한국, 코미디, 드라마, 15세 관람가, 2시간          장진   
21   Assassination, 2015, 한국, 액션, 드라마, 15세 관람가, 2시간...        최동훈   
22        Veteran , 2015, 한국, 액션, 드라마, 15세 관람가, 2시간 4분        류승완   
23   The Maze Runner, 2014, 미국, 액션, SF, 스릴러, 12세 관람...       웨스 볼   
24          D-War, 2007, 한국, 판타지, 액션, 12세 관람가, 1시간 30분        심형래   
25   Architecture 101, 2012, 한국, 로맨스/멜로, 12세 관람가, 1...        이용주   
26         August Rush, 2007, 미국, 드라마, 전체 관람가, 1시간 53분    커스틴 쉐리단   
27   Bruce Almighty, 2003, 미국, 코미디, 판타지, 12세 관람가, 1...      톰 새디악   
28   Lust, Caution, 2007, 미국, 중국, 대만, 홍콩, 로맨스/멜로, 드...         이안   
29   Pulp Fiction, 1994, 미국, 범죄, 코미디, 스릴러, 청소년 관람불가...   쿠엔틴 타란티노   
..                                                 ...        ...   
388  I Saw The Devil, 2010, 한국, 공포, 스릴러, 청소년 관람불가, ...        김지운   
389  The Day After Tomorrow, 2004, 미국, SF, 드라마, 12세...   롤랜드 에머리히   
390  A Dirty Carnival, 2006, 한국, 범죄, 액션, 청소년 관람불가, ...         유하   
391  My PS Partner, 2012, 한국, 로맨틱 코미디, 청소년 관람불가, 1시...        변성현   
392       Blood Rain, 2005, 한국, 스릴러, 청소년 관람불가, 1시간 59분        김대승   
393  The Big Scene, 2005, 한국, 코미디, 스릴러, 15세 관람가, 1시...         장진   
394  My Scary Girl, 2006, 한국, 로맨틱 코미디, 스릴러, 범죄, 청소년...        손재곤   
395      Eagle Eye, 2008, 미국, SF, 액션, 12세 관람가, 1시간 57분   D.J. 카루소   
396  Taken 2, 2012, 프랑스, 액션, 범죄, 드라마, 청소년 관람불가, 1시간...   올리비에 메가턴   
397  Forbidden Quest, 2006, 한국, 코미디, 로맨스/멜로, 12세 관람...        김대우   
398      Poseidon, 2006, 미국, 액션, 드라마, 12세 관람가, 1시간 38분    볼프강 페터젠   
399           2008, 한국, 로맨스/멜로, 드라마, 청소년 관람불가, 1시간 48분        전윤수   
400    The Neighbors, 2012, 한국, 스릴러, 청소년 관람불가, 1시간 50분         김휘   
401  X-Men: First Class, 2011, 미국, 판타지, 액션, 12세 관람가...       매튜 본   
402  Miracle On 1st Street, 2007, 한국, 코미디, 드라마, 15세...        윤제균   
403  The Front Line, 2011, 한국, 전쟁, 드라마, 15세 관람가, 2시...         장훈   
404       Mother, 2009, 한국, 스릴러, 드라마, 청소년 관람불가, 2시간 8분        봉준호   
405       2011, 한국, 로맨틱 코미디, 공포, 드라마, 12세 관람가, 1시간 54분        황인호   
406           Harmony, 2009, 한국, 드라마, 12세 관람가, 1시간 55분        강대규   
407       Hello Ghost, 2010, 한국, 코미디, 12세 관람가, 1시간 51분        김영탁   
408  Harry Potter And The Order Of The Phoenix, 200...    데이빗 예이츠   
409  Harry Potter And The Sorcerer's Stone, 2001, 영...   크리스 콜럼버스   
410                   2008, 한국, 드라마, 청소년 관람불가, 2시간 13분         유하   
411  The Divine Weapon, 2008, 한국, 전쟁, 드라마, 액션, 15세 ...        김유진   
412  Forever The Moment, 2007, 한국, 드라마, 15세 관람가, 2시...        임순례   
413                2012, 한국, 코미디, 드라마, 12세 관람가, 2시간 4분        이석훈   
414  Spider-Man 3, 2007, 미국, 판타지, 액션, 12세 관람가, 2

** 2. df2 : 420~끝까지**

In [44]:
df2 = pd.DataFrame(columns=['DESC', '감독', '배우', '평가자수', '보고싶어요수',
                           '코멘트수', '별점분포'])

for title in url_df['title_url'][420:]:
    response = requests.get("https://watcha.net/mv/{title_url}".format(title_url=title))
    dom = BeautifulSoup(response.content, 'html.parser')
    
    director = dom.select('li.movie-person.director a.kor-name')[1].text # 감독
    actor_list = dom.select('li.movie-person.actor .kor-name')
    actors = []
    for actor in actor_list[1:6:2]:
        actor = actor.text
        actors.append(actor) # 배우 3명
    rate_count = dom.select('div.cumulative-count .rate')[0].text # 평가자 수
    wish_count = dom.select('div.cumulative-count .wish')[0].text # 보고싶어요 수
    review_count = dom.select('div.cumulative-count .review')[0].text # 코메트 수
    desc = dom.select('div.desc')[0].text   # 장르가 포함된 DESC
    rating_dist = dom.select_one('div.rating-chart-wrapper').get('data-distribution') # 분포

    df2.loc[len(df2)] = [desc, director, actors, rate_count, wish_count, review_count, rating_dist]

In [45]:
df2

DESC          감독  \
0       My Teacher, 2005, 한국, 코미디, 액션, 15세 관람가, 2시간 4분         김동원   
1    Mission: Impossible III, 2006, 미국, 액션, 스릴러, 15...  J.J. 에이브럼스   
2                  2010, 한국, 액션, 드라마, 15세 관람가, 1시간 56분          장훈   
3            Tower, 2012, 한국, 액션, 드라마, 12세 관람가, 2시간 1분         김지훈   
4    The War Of Flower, 2006, 한국, 드라마, 코미디, 범죄, 청소년...         최동훈   
5      Les Miserables, 2012, 영국, 드라마, 12세 관람가, 2시간 38분        톰 후퍼   
6    Inception, 2010, 미국, 영국, SF, 스릴러, 액션, 12세 관람가,...    크리스토퍼 놀란   
7    Jeon Woochi : The Taoist Wizard, 2009, 한국, 판타지...         최동훈   
8    War of the Arrows, 2011, 한국, 전쟁, 액션, 청소년 관람불가,...         김한민   
9     Transformers, 2007, 미국, SF, 액션, 12세 관람가, 2시간 15분      마이클 베이   
10   Mission: Impossible: Ghost Protocol, 2011, 미국,...      브래드 버드   
11   Transformers: Dark Of The Moon, 2011, 미국, SF, ...      마이클 베이   
12   The Art Of Seduction, 2005, 한국, 로맨틱 코미디, 15세 관...         오기환   
13   Dancing Princess, 2005, 한국, 로맨스/멜로, 코미디, 12세 관...         박영훈   
14   Very Ordinary Couple (V.O.C.), 2012, 한국, 로맨스/멜...          노덕   
15   Goodbye Mom, 2009, 한국, 드라마, 코미디, 15세 관람가, 1시간 50분         정기훈   
16              As One, 2012, 한국, 드라마, 12세 관람가, 2시간 7분         문현성   
17    Ghost House, 2004, 한국, 코미디, 판타지, 12세 관람가, 2시간 3분         김상진   
18   The Bourne Ultimatum, 2007, 미국, 독일, 액션, 스릴러, 1...     폴 그린그래스   
19           Troy, 2004, 미국, 판타지, 액션, 15세 관람가, 2시간 43분     볼프강 페터젠   
20           300, 2006, 미국, 액션, 판타지, 청소년 관람불가, 1시간 56분      잭 스나이더   
21   Another Public Enemy, 2005, 한국, 드라마, 범죄, 15세 관...         강우석   
22   200 Pounds Beauty, 2006, 한국, 로맨틱 코미디, 드라마, 12세...         김용화   
23   Harry Potter And The Deathly Hallows: Part 2, ...     데이빗 예이츠   
24   The Twilight Saga: New Moon, 2009, 미국, 판타지, 로맨...     크리스 웨이츠   
25   Men In Black II, 2002, 미국, SF, 코미디, 12세 관람가, 1...     베리 소넨필드   
26   A Moment To Remember, 2004, 한국, 로맨스/멜로, 드라마, 1...         이재한   
27   X-Men: The Last Stand, 2006, 미국, SF, 액션, 12세 관...      브렛 래트너   
28   Wanted, 2008, 미국, 독일, 액션, 범죄, 스릴러, 청소년 관람불가, 1...  티무르 베크맘베토브   
29   When Romance Meets Destiny, 2005, 한국, 로맨틱 코미디,...         김현석   
..                                                 ...         ...   
94         Masquerade, 2012, 한국, 드라마, 15세 관람가, 2시간 11분         추창민   
95   Miracle in Cell No.7, 2012, 한국, 코미디, 드라마, 15세 ...         이환경   
96     The Thieves, 2012, 한국, 액션, 범죄, 15세 관람가, 2시간 15분         최동훈   
97   Spirit Of Jeet Keun Do, 2004, 한국, 액션, 드라마, 15세...          유하   
98   71: Into The Fire, 2010, 한국, 전쟁, 액션, 드라마, 12세 ...         이재한   
99    Spider-Man 2, 2004, 미국, 판타지, 액션, 12세 관람가, 2시간 6분       샘 레이미   
100  The Spiriting Away Of Sen And Chihiro, 2001, 일...    미야자키 하야오   
101  The Butterfly Effect, 2004, 미국, 스릴러, SF, 청소년 관...      에릭 브레스   
102  Shutter Island, 2010, 미국, 스릴러, 드라마, 15세 관람가, 2...     마틴 스콜세지   
103  The Truman Show, 1998, 미국, 드라마, 코미디, 15세 관람가, ...       피터 위어   
104  If Only, 2004, 미국, 영국, 로맨스/멜로, 드라마, 15세 관람가, 1...        길 정거   
105  Joint Security Area, 2000, 한국, 드라마, 전쟁, 스릴러, 1...         박찬욱   
106  The Bourne Supremacy, 2004, 미국, 독일, 액션, 스릴러, 1...     폴 그린그래스   
107      Haunters, 2010, 한국, 스릴러, SF, 15세 관람가, 1시간 54분         김민석   
108            Blind, 2011, 한국, 스릴러, 청소년 관람불가, 1시간 51분         안상훈   
109  The School Of Rock, 2003, 미국, 독일, 코미디, 드라마, 전체...   리처드 링클레이터   
110  [500] Days Of Summer, 2009, 미국, 로맨틱 코미디, 드라마, ...        마크 웹   
111  The Shawshank Redemption, 1994, 미국, 드라마, 15세 관...    프랭크 다라본트   
112     Love Letter, 1995, 일본, 로맨스/멜로, 전체 관람가, 1시간 57분      이와이 슌지   
113           2011, 한국, 범죄, 액션, 드라마, 청소년 관람불가, 2시간 13분         윤종빈   
114  The Devil Wears Prada, 2006, 미국, 로맨틱 코미디, 12세 ...     데이빗 프랭클   
115         Moss, 2010, 한국, 스릴러, 범죄, 청소년 관람불가, 2시간 43분         강우석   
116  Real Steel, 2011, 미국, SF, 드라마,액션, 12세 관람가, 2시간 7분        숀 레비   
117  Leon, 1994, 프랑스, 미국, 범죄, 액션, 드라마, 청소년 관람불가, 2시...        뤽 베송   
118  Confession of Murder, 2012, 한국, 스릴러, 액션, 청소년 관...         정병길   
119       Malaton, 2005, 한국, 드라마, 코미디, 전체 관람가, 1시간 55분      

** 3. df418 : 418번 sample(반지의제왕2) - (오류가발생한샘플)직접만듦**

In [83]:
df418 = pd.DataFrame(data=[[
        "The Lord Of The Rings: The Two Towers, 2002, 뉴질랜드, 미국, 판타지, 액션, 12세 관람가, 2시간 57분",
        '피터 잭슨',
        '[일라이저 우드, 비고 모텐슨, 올랜도 볼룸]',
        '384699', '4941', '1190',
        '{"1":1095,"2":5860,"3":1134,"4":12084,"5":4210,"6":51302,"7":22580,"8":121204,"9":25916,"10":139314}'
    ]],
                     columns=['DESC', '감독', '배우', '평가자수', '보고싶어요수', '코멘트수', '별점분포'])

In [84]:
df418

DESC     감독  \
0  The Lord Of The Rings: The Two Towers, 2002, 뉴...  피터 잭슨   

                          배우    평가자수 보고싶어요수  코멘트수  \
0  [일라이저 우드, 비고 모텐슨, 올랜도 볼룸]  384699   4941  1190   

                                                별점분포  
0  {"1":1095,"2":5860,"3":1134,"4":12084,"5":4210...

** 4. df419 : 419번 sample(타이타닉)-실수로 인덱싱못한부분**

In [81]:
df419 = pd.DataFrame(columns=['DESC', '감독', '배우', '평가자수', '보고싶어요수',
                           '코멘트수', '별점분포'])

for title in url_df['title_url'][419:420]:
    response = requests.get("https://watcha.net/mv/{title_url}".format(title_url=title))
    dom = BeautifulSoup(response.content, 'html.parser')
    
    director = dom.select('li.movie-person.director a.kor-name')[1].text # 감독
    actor_list = dom.select('li.movie-person.actor .kor-name')
    actors = []
    for actor in actor_list[1:6:2]:
        actor = actor.text
        actors.append(actor) # 배우 3명
    rate_count = dom.select('div.cumulative-count .rate')[0].text # 평가자 수
    wish_count = dom.select('div.cumulative-count .wish')[0].text # 보고싶어요 수
    review_count = dom.select('div.cumulative-count .review')[0].text # 코메트 수
    desc = dom.select('div.desc')[0].text   # 장르가 포함된 DESC
    rating_dist = dom.select_one('div.rating-chart-wrapper').get('data-distribution') # 분포

    df419.loc[len(df419)] = [desc, director, actors, rate_count, wish_count, review_count, rating_dist]

In [82]:
df419

DESC       감독  \
0  Titanic, 1997, 미국, 로맨스/멜로, 드라마, 15세 관람가, 3시간 15분  제임스 카메론   

                                  배우     평가자수  보고싶어요수   코멘트수  \
0  [레오나르도 디카프리오, 케이트 윈슬렛, 글로리아 스튜어트]  654,853  14,918  2,895   

                                                별점분포  
0  {"1":668,"2":3526,"3":790,"4":8881,"5":3769,"6...

### 완성된 추가 feature dataframe = df
 * append

In [86]:
df = df1.append([df418, df419, df2], ignore_index=True)
df[400:]

DESC          감독  \
400    The Neighbors, 2012, 한국, 스릴러, 청소년 관람불가, 1시간 50분          김휘   
401  X-Men: First Class, 2011, 미국, 판타지, 액션, 12세 관람가...        매튜 본   
402  Miracle On 1st Street, 2007, 한국, 코미디, 드라마, 15세...         윤제균   
403  The Front Line, 2011, 한국, 전쟁, 드라마, 15세 관람가, 2시...          장훈   
404       Mother, 2009, 한국, 스릴러, 드라마, 청소년 관람불가, 2시간 8분         봉준호   
405       2011, 한국, 로맨틱 코미디, 공포, 드라마, 12세 관람가, 1시간 54분         황인호   
406           Harmony, 2009, 한국, 드라마, 12세 관람가, 1시간 55분         강대규   
407       Hello Ghost, 2010, 한국, 코미디, 12세 관람가, 1시간 51분         김영탁   
408  Harry Potter And The Order Of The Phoenix, 200...     데이빗 예이츠   
409  Harry Potter And The Sorcerer's Stone, 2001, 영...    크리스 콜럼버스   
410                   2008, 한국, 드라마, 청소년 관람불가, 2시간 13분          유하   
411  The Divine Weapon, 2008, 한국, 전쟁, 드라마, 액션, 15세 ...         김유진   
412  Forever The Moment, 2007, 한국, 드라마, 15세 관람가, 2시...         임순례   
413                2012, 한국, 코미디, 드라마, 12세 관람가, 2시간 4분         이석훈   
414  Spider-Man 3, 2007, 미국, 판타지, 액션, 12세 관람가, 2시간 19분       샘 레이미   
415  My Tutor Friend, 2003, 한국, 로맨틱 코미디, 12세 관람가, 1...         김경형   
416  The Chaser, 2008, 한국, 스릴러, 범죄, 액션, 청소년 관람불가, 2...         나홍진   
417  Pirates Of The Caribbean: At World's End, 2007...     고어 버빈스키   
418  The Lord Of The Rings: The Two Towers, 2002, 뉴...       피터 잭슨   
419   Titanic, 1997, 미국, 로맨스/멜로, 드라마, 15세 관람가, 3시간 15분     제임스 카메론   
420     My Teacher, 2005, 한국, 코미디, 액션, 15세 관람가, 2시간 4분         김동원   
421  Mission: Impossible III, 2006, 미국, 액션, 스릴러, 15...  J.J. 에이브럼스   
422                2010, 한국, 액션, 드라마, 15세 관람가, 1시간 56분          장훈   
423          Tower, 2012, 한국, 액션, 드라마, 12세 관람가, 2시간 1분         김지훈   
424  The War Of Flower, 2006, 한국, 드라마, 코미디, 범죄, 청소년...         최동훈   
425    Les Miserables, 2012, 영국, 드라마, 12세 관람가, 2시간 38분        톰 후퍼   
426  Inception, 2010, 미국, 영국, SF, 스릴러, 액션, 12세 관람가,...    크리스토퍼 놀란   
427  Jeon Woochi : The Taoist Wizard, 2009, 한국, 판타지...         최동훈   
428  War of the Arrows, 2011, 한국, 전쟁, 액션, 청소년 관람불가,...         김한민   
429   Transformers, 2007, 미국, SF, 액션, 12세 관람가, 2시간 15분      마이클 베이   
..                                                 ...         ...   
514        Masquerade, 2012, 한국, 드라마, 15세 관람가, 2시간 11분         추창민   
515  Miracle in Cell No.7, 2012, 한국, 코미디, 드라마, 15세 ...         이환경   
516    The Thieves, 2012, 한국, 액션, 범죄, 15세 관람가, 2시간 15분         최동훈   
517  Spirit Of Jeet Keun Do, 2004, 한국, 액션, 드라마, 15세...          유하   
518  71: Into The Fire, 2010, 한국, 전쟁, 액션, 드라마, 12세 ...         이재한   
519   Spider-Man 2, 2004, 미국, 판타지, 액션, 12세 관람가, 2시간 6분       샘 레이미   
520  The Spiriting Away Of Sen And Chihiro, 2001, 일...    미야자키 하야오   
521  The Butterfly Effect, 2004, 미국, 스릴러, SF, 청소년 관...      에릭 브레스   
522  Shutter Island, 2010, 미국, 스릴러, 드라마, 15세 관람가, 2...     마틴 스콜세지   
523  The Truman Show, 1998, 미국, 드라마, 코미디, 15세 관람가, ...       피터 위어   
524  If Only, 2004, 미국, 영국, 로맨스/멜로, 드라마, 15세 관람가, 1...        길 정거   
525  Joint Security Area, 2000, 한국, 드라마, 전쟁, 스릴러, 1...         박찬욱   
526  The Bourne Supremacy, 2004, 미국, 독일, 액션, 스릴러, 1...     폴 그린그래스   
527      Haunters, 2010, 한국, 스릴러, SF, 15세 관람가, 1시간 54분         김민석   
528            Blind, 2011, 한국, 스릴러, 청소년 관람불가, 1시간 51분         안상훈   
529  The School Of Rock, 2003, 미국, 독일, 코미디, 드라마, 전체...   리처드 링클레이터   
530  [500] Days Of Summer, 2009, 미국, 로맨틱 코미디, 드라마, ...        마크 웹   
531  The Shawshank Redemption, 1994, 미국, 드라마, 15세 관...    프랭크 다라본트   
532     Love Letter, 1995, 일본, 로맨스/멜로, 전체 관람가, 1시간 57분      이와이 슌지   
533           2011, 한국, 범죄, 액션, 드라마, 청소년 관람불가, 2시간 13분         윤종빈   
534  The Devil Wears Prada, 2006, 미국, 로맨틱 코미디, 12세 ...     데이빗 프랭클   
535         Moss, 2010, 한국, 스릴러, 범죄, 청소년 관람불가, 2시간 43분         강우석   
536  Real Steel, 2011, 미국, SF, 드라마,액션, 12세 관람가, 2시간 7분        숀 레비   
537  Leon, 1994, 프랑스, 미국, 범죄, 액션, 드라마, 청소년 관람불가, 2시...        뤽 베송   
538  Confession of Murder, 2012, 한국, 스릴러, 액션, 청소년 관...         정병길   
539       Malaton, 2005, 한국, 드라마, 코미디, 전체 관람가, 1시간 55분      

---

### 이동진 평가 dataframe = lee_df
* 작업시간 약 5분

In [13]:
lee_df = pd.DataFrame(columns=['name', '이동진 별점', '이동진 코멘트'])

for code in url_df['code']:
    json_response = requests.get(
        "https://watcha.net/comment/list?unique_id={code_list}&start_index=0&count=5&type=like".format(
        code_list = code))
    comment_dict = json.loads(json_response.text)
    
    lee = comment_dict.get('data')[0].get('username')
    lee_rating = comment_dict.get('data')[0].get('rating')
    lee_comment = comment_dict.get('data')[0].get('text')
    
    lee_df.loc[len(lee_df)] = [lee, lee_rating, lee_comment]


In [14]:
lee_df

name  이동진 별점  \
0                 이동진 평론가     4.5   
1                     김수빈     3.0   
2                 Sloth K     5.0   
3                     송재한     2.0   
4                     김기광     NaN   
5                      영웅     0.5   
6                  keeper     3.0   
7                     허진진     4.5   
8                      히스     3.5   
9                     김민지     4.0   
10                    김용현     4.0   
11                    류주니     4.0   
12                이동진 평론가     4.0   
13                    이동호     3.0   
14                   워노워노     3.5   
15                   워노워노     3.5   
16                이동진 평론가     3.5   
17                    김태균     4.5   
18                    김태균     1.5   
19                    김경찬     5.0   
20                    문성희     5.0   
21                    최영생     4.5   
22                이동진 평론가     3.5   
23                    박정서     3.5   
24                    최창원     0.5   
25                     준희     5.0   
26                    이관민     2.0   
27          Jae Bong Choi     4.0   
28                    이상혁     4.0   
29                    임기원     3.0   
..                    ...     ...   
514                   박주영     5.0   
515                kkrrrr     2.0   
516                   심세훈     3.5   
517                   김미은     3.0   
518               이동진 평론가     1.5   
519                   신현영     5.0   
520                  JM.Y     5.0   
521                  가키레나     5.0   
522                    원뉴     4.5   
523                  클린트옹     5.0   
524                   이다예     0.5   
525           April Byeon     4.5   
526                   KCH     4.0   
527                    NY     1.0   
528               이동진 평론가     3.0   
529               이동진 평론가     3.5   
530               이동진 평론가     4.5   
531       christmascactus     5.0   
532               이동진 평론가     4.5   
533  Damian Jeon DongHyun     5.0   
534                   jsp     4.5   
535               이동진 평론가     3.5   
536                   최영훈     5.0   
537                    한준     5.0   
538                niknik     5.0   
539                   김준형     4.0   
540                   신한나     4.0   
541                   신현영     4.5   
542                   이환희     3.0   
543                   이알미     5.0   

                                               이동진 코멘트  
0      &#x27;모두가 대통령의 사람들&#x27;과 함께 오래도록 거론될 모범적 언론영화.  
1    영화 내내 드는 생각은 김강우는 뭐때문에 저렇게까지 열심히뛰지? 그리고 박성웅이 김...  
2                        선라이즈의 동화,\n선셋의 낭만,\n미드나잇의 시간.  
3                                     이랬다가 저랬다가 왔다 갔다.  
4    영화보다 대단한게 한국 광고 회사라는 거다. 몇 백만의 관객에게 놀란의 영화라 속였...  
5            시사회 보고 왔는데 내 인생에서 두시간을 낭비함... 아 3분 정도는 건짐  
6                                            두시간짜리 예고편  
7    난 우리나라기업에게도 이런식의 직원채용이필요하다고 절실히느꼈다\n연륜에서 나오는 지...  
8    겉모습이 중요하지 않다고 말하면서 결정적인 순간마다 등장하는 &#x27;젊고 잘생긴...  
9    초반은 살짝지루하나 강동원이 잘생겼으며, 전체적으로 단조로운 구성이지만 강동원이 잘...  
10   흔히 물의를 일으킨 연기자들은 이렇게 말한다. &quot;더 좋은 연기로 찾아뵙겠다...  
11                                              기분나쁜영화  
12                  하나를 위한 전체.  세상에서 가장 낙천적인 (SF)재난영화.  
13   미드나잇 인 파리, 어바웃 타임 그리고 시간 여행자의 아내까지 레이첼 맥아담스는 진...  
14                                      강동원(35) 처녀귀신 역  
15                         C급 현실성 B급 스토리 A급 배우들 S급 영상미  
16                      두 주인공이 처음 직접 대면하게 될 때의 굉장한 박력.  
17   영화의 대사를 빌려서: &quot;Yes, I am satisfied with my...  
18            그녀는 그렇게 Her (2013)의 사만다 (Samantha)가 되었다.  
19                   ‘세상에서 가장 예쁜 여인과 키스하기’ 항목을 지울 때...  
20                                     원빈 보려다 신하균한테 홀림  
21   수 많은 염석진은 아직 살아있고\n잊지 말아달라던 사람들은 잊혀졌고\n부서진 집을 ...  
22                       심플하고 힘있게 치고 달리는 영화가 선사하는 통쾌함.  
23                                      생각보다 놀란 민호의 비중  
24                  축하드립니다 드디어 왓챠에 별점 반 개 기능이 추가되었습니다!  
25     첫사랑, 그 설레고 아련했던 사랑의 감정 그리고 풋풋하고 찌질했던 우리 모두의 이야기  
26           감동을 위해서라면 최소한의 개연성마저 과감히 생략하는 감독의 기백이 돋보임  
27                        신을 흑인으로 설정했다는데 더 높은 점수를 주고싶다  
28                                  엑기스만 먼저 봤던 나를 반성한다  
29              

---

### 2개의 dataframe => csv파일로 저장
* df, lee_df

In [88]:
path='C:/Users/JKEUN/ipython notebook/project-02-watcha/resource/'
df.to_csv(path+'raw_df2.csv', index=False, encoding='utf8')

In [ ]:
path='C:/Users/JKEUN/ipython notebook/project-02-watcha/resource/'
lee_df.to_csv(path+'lee_df.csv', index=False, encoding='utf8')